core/src/main/java/binnie/core/genetics/Tolerance.java

```diff
@@ -29,9 +33,19 @@ public static Tolerance get(final forestry.api.genetics.EnumTolerance tol) {
 		return get(tol).canTolerate(base, test);
 	}
 
+	@Nullable
+	private static Pattern PATTERN;
+
+	private static Pattern GetPattern() {
```

Comment:
Method names should start with a lowercase letter.

Actually I think this should just be created statically instead of using a singleton getter, since it is not very expensive to create and does not pull in lots of other classes.
```java
private static Pattern PATTERN = Pattern.compile("_", Pattern.LITERAL);
```

Reply:
In this case it is not. At first version I do like you describe. But after mc run, it crushed with NullPointException in 
```
@Override
	public String toString() {
		return GetPattern().matcher(super.toString().toLowerCase(Locale.ENGLISH)).replaceAll(StringUtils.EMPTY);
}
```
Where is some issue with object creating (initial). At moment then toString invoked PATTERN will be null.

Reply:
Hm enums are weird, I guess they are being constructed before static. This is fine then.